examples/pytorch/rgcn/experimental/README.md

```diff
@@ -1,62 +1,64 @@
-
 To train RGCN, it has four steps:
 
-### Step 0: set IP configuration file.
+### Step 0: Setup a Distributed File System 
+* You may skip this step if your cluster already has folder(s) synchronized across machines. 
 
-User need to set their own IP configuration file before training. For example, if we have four machines in current cluster, the IP configuration
-could like this:
+To perform distributed training, files and codes need to be accessed across multiple machines. A distributed file system would perfectly handle the job (i.e., NFS, Ceph). 
 
+Here is an example of how to setup NFS. First, install essential libs on the storage server
 ```bash
-172.31.0.1
-172.31.0.2
-172.31.0.3
-172.31.0.4
-```
+sudo apt-get install nfs-kernel-server
+``` 
 
-Users need to make sure that the master node (node-0) has right permission to ssh to all the other nodes.
+Then create a folder for sharing and (optional) remove all restrictions
+```bash
+sudo mkdir -p /NFS
+# Change ownership and remove restrictions
+sudo chown -R nobody:nogroup /NFS
+sudo chmod 755 /NFS
+```
 
-### Step 1: partition the graph.
+We assume that the all servers are under a subnet with ip range `192.168.0.0` to `192.168.255.255`. The exports configuration needs to be modifed to 
 
-The example provides a script to partition some builtin graphs such as ogbn-mag graph.
-If we want to train RGCN on 4 machines, we need to partition the graph into 4 parts.
+```bash
+sudo vim /etc/exports
+# add the following line 
+/NFS  192.168.0.0/16(rw,sync,no_subtree_check)
+```
 
-In this example, we partition the ogbn-mag graph into 4 parts with Metis. The partitions are balanced with respect to
-the number of nodes, the number of edges and the number of labelled nodes.
+The server's internal ip can be checked  via `ifconfig` or `ip`. If the ip does not begin with `192.168`, then you may use
 ```bash
-python3 partition_graph.py --dataset ogbn-mag --num_parts 4 --balance_train --balance_edges
+# for ip range 10.0.0.0 – 10.255.255.255	
+/NFS  10.0.0.0/8(rw,sync,no_subtree_check)
+# for ip range 172.16.0.0 – 172.31.255.255	
+/NFS  172.16.0.0/12(rw,sync,no_subtree_check)
 ```
 
-This script generates partitioned graphs and store them in the directory called `data`.
+For configraution details, please refer to [NFS ArchWiki](https://wiki.archlinux.org/index.php/NFS).
+
+<details><summary>If above setup does not work,  DGL provides a backup solution</summary>
+<p>
 
-### Step 2: copy the partitioned data to the cluster
-DGL provides a script for copying partitioned data to the cluster. Before that, copy the training script to a local folder:
+
+DGL provides a script for copying partitioned data to the cluster. 
+Before that, copy the training script to a local folder:
 
 
 ```bash
 mkdir ~/dgl_code
-cp ~/dgl/examples/pytorch/rgcn/experimental/entity_classify_dist.py ~/dgl_code
+cp /home/ubuntu/dgl/examples/pytorch/rgcn/experimental/entity_classify_dist.py ~/dgl_code
```

Comment:
why do you change it to use `/home/ubuntu`?

Reply:
it is a typo. Fixed